pkg/controller/nodes/task/resourcemanager/resourcemanager.go

```diff
@@ -12,23 +12,36 @@ import (
 )
 
 type TokenPrefix string
+type Token string
 
 const execUrnPrefix = "ex"
 const execUrnSeparator = ":"
 const tokenNamespaceSeparator = "-"
 
-func (t TokenPrefix) append(s string) string {
-	return fmt.Sprintf("%s%s%s", t, tokenNamespaceSeparator, s)
+// appending the actual token string to the prefix using '-' separator
+func (t TokenPrefix) append(tokenWithoutPrefix string) Token {
+	return Token(fmt.Sprintf("%s%s%s", t, tokenNamespaceSeparator, tokenWithoutPrefix))
 }
 
-func composeExecutionUrn(id *core.TaskExecutionIdentifier) string {
-	return execUrnPrefix + execUrnSeparator + id.GetNodeExecutionId().GetExecutionId().GetProject() +
-		execUrnSeparator + id.GetNodeExecutionId().GetExecutionId().GetDomain() + execUrnSeparator + id.GetNodeExecutionId().GetExecutionId().GetName()
+// extending the prefix using ':' separator
+func (t TokenPrefix) extend(prefixPart string) TokenPrefix {
```

Comment:
When do you use extend vs append? the docs are not clear to me of the implications...

Reply:
Refactored. Now I have TokenPrefix.extend() and Token.prepend(). Is it clearer?